jerry-core/parser/js/js-scanner.c

```diff
@@ -3193,16 +3197,14 @@ scanner_scan_all (parser_context_t *context_p, /**< context */
     }
     PARSER_CATCH
     {
-      JERRY_ASSERT (context_p->error == PARSER_ERR_NO_ERROR);
-
-      while (scanner_context.active_literal_pool_p != NULL)
-      {
-        scanner_literal_pool_t *literal_pool_p = scanner_context.active_literal_pool_p;
+      JERRY_ASSERT (context_p->error == PARSER_ERR_NO_ERROR || context_p->error == PARSER_ERR_OUT_OF_MEMORY);
```

Comment:
I realized something. Iff we are inside the catch, how can this be `PARSER_ERR_NO_ERROR`?

Reply:
True, this case cannot happen.